### PR TITLE
feat: handle sold out projects

### PR DESCRIPTION
**Pull request by flagrede**
## Description

Closes: regen-network/regen-registry#1518

- Create [GradientBadge](https://github.com/regen-network/regen-web/pull/1773/files#diff-f02a7adcd8babe73f60471d7727cec57b7fa3993fa3e555d0362e24d71c6e663) component
- Add badge + tooltip to [ProjectCard](https://github.com/regen-network/regen-web/pull/1773/files#diff-f17460cd760307f1e339aed0bfd7a895e11357650c6bb0b0d74c3d8ea8329d9f)
- Add tooltips, icons and sold out badge (whenever needed) to [ProjectBatchTotals](https://github.com/regen-network/regen-web/pull/1773/files#diff-aeb061bfdd2530072327711082cef7456b6eead30b4cc6d379899e8c0bfba509) (statistics on project details page)
- add [sanity query](https://github.com/regen-network/regen-web/pull/1773/files#diff-bda903a60a377d3fe4ff6cf47164d5d984f5999d8ea61f9a55e35e6aebaa3815) to get sold-out projects ids list
- remove `track` dependency from `ProjectCard` to fix storybook story

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] provided a link to the relevant issue or specification
- [ ] provided instructions on how to test
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed
- [ ] once the PR is closed, set up backport PRs for `redwood`, `hambach` and optionally `master` (see below)

#### Setting up backport PRs

After merging your PR to `dev`, set up backports by doing the following:

1. If your branch does not have merge commits, add the following comment to
   your PR:

- If PR is a new feature: `@Mergifyio backport redwood hambach`.
- If PR is a bug fix: `@Mergifyio backport redwood hambach master`.

2. If your branch does have merge commits:

   a. Pull latest `dev`, `hambach` and `redwood` branches

   b. Create new branches for backports and merge `dev` (replace `<PR#>` with your PR #)

   ```
   git checkout -b hambach-backport-<PR#> hambach
   git merge dev
   git push origin hambach-backport-<PR#>
   git checkout -b redwood-backport-<PR#> redwood
   git merge dev
   git push origin redwood-backport-<PR#>`
   ```

   c. Open new PRs in regen-web targeting `hambach` and `redwood`, respectively.

   d. If PR is a bug fix repeat the same process for `master` branch.

### How to test

1. https://deploy-preview-1773--regen-registry.netlify.app/

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by flagrede** [comment]
@erikalogie you can test the sold-out projects improvement here: https://deploy-preview-1773--regen-registry.netlify.app/
(available on all `ProjectCard`and project details statistics section)

**Comment by erikalogie** [comment]
@flagrede this looks great! Only small nit is that probably we should spell it "credits tradable" not "credits tradeable" everywhere to be consistent. Nice work!

**Comment by blushi** [comment]
@flagrede should we have a PR on regen-sanity?

**Comment by flagrede** [comment]
@blushi forgot to do it yesterday, here it is: https://github.com/regen-network/regen-sanity/pull/25

File: web-registry/src/components/templates/ProjectDetails/ProjectDetails.tsx [hunk 1]
```diff
@@ -61,6 +62,15 @@ function ProjectDetails(): JSX.Element {
...
     sanityProjectPageData?.allProjectPage?.[0]?.gettingStartedResourcesSection;
 
+  const { data: sanitySoldOutProjects } = useAllSoldOutProjectsQuery({
+    client: sanityClient,
+  });
+
+  const soldOutProjectsIds =
+    sanitySoldOutProjects?.allSoldOutProjects?.[0]?.soldOutProjectsList?.map(
+      project => String(project?.projectId),
+    ) ?? [];
```
Comment (blushi) [inline web-registry/src/components/templates/ProjectDetails/ProjectDetails.tsx:72]: maybe we can extract this part into a hook that could be re-used in `ProjectCardsSection` too?

Comment (blushi) [reply web-registry/src/components/templates/ProjectDetails/ProjectDetails.tsx:72]: why aren't we using react-query here?

File: web-registry/src/pages/Projects/Projects.tsx [hunk 2]
```diff
@@ -43,6 +50,14 @@ export const Projects: React.FC<React.PropsWithChildren<unknown>> = () => {
     sanityProjectsPageData?.allProjectsPage?.[0]
       ?.gettingStartedResourcesSection;
 
+  const { data: sanitySoldOutProjects } = useAllSoldOutProjectsQuery({
+    client: sanityClient,
+  });
+  const soldOutProjectsIds =
+    sanitySoldOutProjects?.allSoldOutProjects?.[0]?.soldOutProjectsList?.map(
+      project => String(project?.projectId),
+    ) ?? [];
```
Comment (blushi) [inline web-registry/src/pages/Projects/Projects.tsx:59]: ditto

**Review by blushi — COMMENTED** [review]
looks good, just a few nits

File: web-registry/src/components/templates/ProjectDetails/ProjectDetails.tsx (hunk 1 above)
Comment (flagrede) [reply web-registry/src/components/templates/ProjectDetails/ProjectDetails.tsx:72]: I actually created the query and everything and up not using it 😅, thanks for catching that.

**Review by blushi — APPROVED** [review]

**Comment by flagrede** [comment]
@Mergifyio backport redwood

**Pull request merged**